src/Data/Connection/AbstractConnectionResolver.php

```diff
@@ -593,7 +593,7 @@ public function get_nodes() {
 		$ids = array_slice( $ids, 0, $this->query_amount, true );
 
 		// If pagination is going backwards, revers the array of IDs
-		$ids = ! empty( $this->args['before'] ) ? array_reverse( $ids ) : $ids;
+		$ids = ! empty( $this->args['last'] ) ? array_reverse( $ids ) : $ids;
```

Comment:
Do you also need `! empty( $this->args['before'] )` here? Can 'before' be specified for backward pagination without 'last'?

Reply:
@markkelnar No. It was a mistake for me to have reversed order by `before` in the first place. 

Let's say we have a list like so: `[ a, b, c, d, e, f, g, h, i, j ]`

If I ask for `first: 3`

I should get `a, b, c`

If I ask for `first: 3, before: h`, I would get `a,b,c` because they are the `first 3` items in the list, and they are `before h` in the list.

If I ask for `first: 3, after: h` I should get: `i, j` as they are the first (up to) 3 items in the list, and they are `after h`

If I ask for `last: 3` the list would come to us (from the database) as `[j, i, h, g, f, e, d, c, b, a]` so we need to reverse as it, as the items we need to return are `i, j`. So, we slice the array, to `j, i`, then reverse it to `i, j` to adhere to the Relay spec that needs the startCursor (`i`) to be the first item. 

Now, we can page backward: `last: 2, before: i` and we slice the array as `h, g`, reverse it, and return `g, h` with `g` as the startCursor so we can keep going backward. 

Drafting this response helped me with writing some tests for this PR! 😄